plugin/cache/cache.go

```diff
@@ -76,21 +75,39 @@ func key(qname string, m *dns.Msg, t response.Type, do bool) (bool, uint64) {
 	return true, hash(qname, m.Question[0].Qtype, do)
 }
 
-var one = []byte("1")
-var zero = []byte("0")
+type fnv64 uint64
```

Comment:
This needs some documentation on what's being done here. Is there any (semi) official hash algo that we can use? And why isn't fnv in the std library more efficient?

Reply:
I'll document it, this is the same `fnv` function as the stdlib, but has a WriteByte and WriteString method.